flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/calcite/FlinkRelBuilder.scala

```diff
@@ -186,6 +195,44 @@ class FlinkRelBuilder(
     push(relNode)
     this
   }
+
+  def compactScan(
+      identifier: ObjectIdentifier,
+      catalogTable: ResolvedCatalogTable,
+      isTemporary: Boolean,
+      compactOptions: util.Map[String, String]): RelBuilder = {
+    val flinkContext = context.unwrap(classOf[FlinkContext])
+    val config = flinkContext.getTableConfig.getConfiguration
+
+    val hints = new util.ArrayList[RelHint]
```

Comment:
Can you just use `CatalogSourceTable` from `FlinkCalciteCatalogReader`? (You can refer to `RelBuilder`) Then you can create a `ToRelContext` for it.